src/lib/libstrings.js

```diff
@@ -322,23 +322,22 @@ addToLibrary({
 #if ASSERTIONS
     assert(ptr % 2 == 0, 'Pointer passed to UTF16ToString must be aligned to two bytes!');
 #endif
+    var idx = {{{ getHeapOffset('ptr', 'u16') }}};
+    var maxIdx = idx + maxBytesToRead / 2;
 #if TEXTDECODER
-    var endPtr = ptr;
     // TextDecoder needs to know the byte length in advance, it doesn't stop on
     // null terminator by itself.
     // Also, use the length info to avoid running tiny strings through
     // TextDecoder, since .subarray() allocates garbage.
-    var idx = endPtr >> 1;
-    var maxIdx = idx + maxBytesToRead / 2;
+    var endIdx = idx;
     // If maxBytesToRead is not passed explicitly, it will be undefined, and this
     // will always evaluate to true. This saves on code size.
-    while (!(idx >= maxIdx) && HEAPU16[idx]) ++idx;
-    endPtr = idx << 1;
+    while (!(endIdx >= maxIdx) && HEAPU16[endIdx]) ++endIdx;
 
 #if TEXTDECODER != 2
-    if (endPtr - ptr > 32 && UTF16Decoder)
+    if (endIdx - idx > 16 && UTF16Decoder)
```

Comment:
Is the 16 to 32 change here because the diff here is between two 16-bit indexes now, but was between bytes index before?

Can we add a comment here?  Something like: `For short (<16 code points) strings the call to `UTF16Decoder.decode is not worth it?` ... assuming that is what is going on here?

Reply:
Yeah, bytes. On my phone now but I think there was an existing comment to that extent above?

Reply:
Ah no, that was just in the UTF-8 decoder. Will do.